conans/client/api/conan_api.py

```diff
@@ -92,26 +92,30 @@ def load_remotes(self, remote_name=None, update=False, check_updates=False):
 
 
 def api_method(f):
+    """Useful decorator to manage Conan API methods"""
+    @functools.wraps(f)
     def wrapper(api, *args, **kwargs):
-        try:  # getcwd can fail if Conan runs on an unexisting folder
+        try:  # getcwd can fail if Conan runs on an non-existing folder
             old_curdir = os.getcwd()
         except EnvironmentError:
             old_curdir = None
         try:
             api.create_app()
             with environment_append(api.app.cache.config.env_vars):
                 return f(api, *args, **kwargs)
-        except Exception as exc:
-            msg = exception_message_safe(exc)
-            try:
-                api.out.error("{} ({})".format(str(exc.__class__.__name__), msg))
-            except BaseException:
-                pass
-            raise
+        except (NotFoundException, PackageNotFoundException) as e:
+            # This exception must be caught manually due to a server inconsistency:
+            # Artifactory API returns an empty result if the recipe doesn't exist, but
+            # Conan Server returns a 404. This probably should be fixed server side,
+            # but in the meantime we must handle it here
+            raise ConanException(f"NotFoundError: {str(e)}")
+        except ConanConnectionError as e:
```

Comment:
This re-raise as ConanExceptions of all exceptions reads a bit weird, specially as it is not adding more context data, just the name of the exception, and that could be encoded directly in the exception?